csharp/src/Drivers/Apache/Hive2/HiveServer2Reader.cs

```diff
@@ -79,14 +81,34 @@ public HiveServer2Reader(
 
         public async ValueTask<RecordBatch?> ReadNextRecordBatchAsync(CancellationToken cancellationToken = default)
         {
+            // All records have been exhausted
             if (_statement == null)
             {
                 return null;
             }
 
-            var request = new TFetchResultsReq(_statement.OperationHandle, TFetchOrientation.FETCH_NEXT, _statement.BatchSize);
-            TFetchResultsResp response = await _statement.Connection.Client.FetchResults(request, cancellationToken);
+            // This is the initial case, before the first fetch
+            _fetchResultResponseTask ??= FetchNext(_statement, cancellationToken);
```

Comment:
Should the first fetch task be started immediately in the constructor? I don't imagine it would make a difference from a performance perspective, but it feels like it would be a little cleaner.

Reply:
Moved the initial pre-fetch to the initializer.